includes/admin/reports/class-wc-admin-report.php

```diff
@@ -322,30 +335,103 @@ public function get_order_report_data( $args = array() ) {
 
 		$query          = apply_filters( 'woocommerce_reports_get_order_report_query', $query );
 		$query          = implode( ' ', $query );
-		$query_hash     = md5( $query_type . $query );
-		$cached_results = get_transient( strtolower( get_class( $this ) ) );
 
 		if ( $debug ) {
 			echo '<pre>';
 			wc_print_r( $query );
 			echo '</pre>';
 		}
 
-		if ( $debug || $nocache || false === $cached_results || ! isset( $cached_results[ $query_hash ] ) ) {
-			static $big_selects = false;
-			// Enable big selects for reports, just once for this session
-			if ( ! $big_selects ) {
-				$wpdb->query( 'SET SESSION SQL_BIG_SELECTS=1' );
-				$big_selects = true;
+		if ( $debug || $nocache ) {
+			self::enable_big_selects();
+
+			$result = apply_filters( 'woocommerce_reports_get_order_report_data', $wpdb->$query_type( $query ), $data );
```

Comment:
Could this section be refactored to only run the query/filter part once without code duplication?

i.e. get the hash and cached result and run the query if null or if we want to skip cache.

Reply:
Nope it's either $result = $this->get_cached_query( $query_hash ) in the if or 2 lines of code duplication else I'm effrayed it's gonna result in way more code duplication because I'll have to put more if (!$debug && !$nocache) under the initial if